core/2d/Label.cpp

```diff
@@ -1491,22 +1493,35 @@ void Label::enableBold()
 
 void Label::enableUnderline()
 {
-    // remove it, just in case to prevent adding two or more
-    if (!_underlineNode)
+    if (_underlineEnabled)
     {
-        _underlineNode = DrawNode::create();
-        _underlineNode->setGlobalZOrder(getGlobalZOrder());
-        addChild(_underlineNode, 100000);
-        _contentDirty = true;
+        return;
+    }
+    _underlineEnabled = true;
+    _contentDirty     = true;
+    if (!_lineDrawNode)
+    {
+        _lineDrawNode = DrawNode::create();
+        _lineDrawNode->setGlobalZOrder(getGlobalZOrder());
+        _lineDrawNode->properties.setFactor(_lineDrawNode->properties.getFactor() * 2.0f);
+        addChild(_lineDrawNode, 100000);
```

Comment:
Magic number?

Reply:
Makes the line smaller "text behind a strikethrough font" is reading easier.
And an "underline line" should not be bigger as a "strikethrough line"